--- a/Makefile
+++ b/Makefile
@@ -2220,12 +2220,13 @@ strip:
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
 
 install:
-	mkdir -p $(MODDESTDIR)
+	#mkdir -p $(MODDESTDIR)
 	install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)
 	/sbin/depmod -a ${KVER}
 
 uninstall:
-	rm -rfv $(MODDESTDIR)
+	#rm -rfv $(MODDESTDIR)
+	rm -f $(MODDESTDIR)/$(MODULE_NAME).ko
 	/sbin/depmod -a ${KVER}
 
 backup_rtlwifi:
@@ -2270,18 +2271,19 @@ config_r:
 .PHONY: modules clean
 
 clean:
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd) clean
 	#$(MAKE) -C $(KSRC) M=$(shell pwd) clean
-	cd hal ; rm -fr */*/*/*.mod.c */*/*/*.mod */*/*/*.o */*/*/.*.cmd */*/*/*.ko
-	cd hal ; rm -fr */*/*.mod.c */*/*.mod */*/*.o */*/.*.cmd */*/*.ko
-	cd hal ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
-	cd hal ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
-	cd core ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
-	cd core ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
-	cd os_dep/linux ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
-	cd os_dep ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
-	cd platform ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
-	rm -fr Module.symvers ; rm -fr Module.markers ; rm -fr modules.order
-	rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~
-	rm -fr .tmp_versions
+	#cd hal ; rm -fr */*/*/*.mod.c */*/*/*.mod */*/*/*.o */*/*/.*.cmd */*/*/*.ko
+	#cd hal ; rm -fr */*/*.mod.c */*/*.mod */*/*.o */*/.*.cmd */*/*.ko
+	#cd hal ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
+	#cd hal ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
+	#cd core ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
+	#cd core ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
+	#cd os_dep/linux ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
+	#cd os_dep ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
+	#cd platform ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
+	#rm -fr Module.symvers ; rm -fr Module.markers ; rm -fr modules.order
+	#rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~
+	#rm -fr .tmp_versions
 endif
